Packages/ClientRuntime/Sources/Networking/Http/CRT/CRTClientEngine.swift

```diff
@@ -109,43 +87,43 @@ public class CRTClientEngine: HttpClientEngine {
         }
     }
     
-    public func makeHttpRequestStreamOptions(_ request: SdkHttpRequest) -> (HttpRequestOptions, Future<HttpResponse>) {
-        let future = Future<HttpResponse>()
-        let crtRequest = request.toHttpRequest(bufferSize: windowSize)
+    public func makeHttpRequestStreamOptions(_ request: SdkHttpRequest, _ continuation: StreamContinuation) -> HttpRequestOptions {
         let response = HttpResponse()
-        
+        let crtRequest = request.toHttpRequest(bufferSize: windowSize)
         let streamReader: StreamReader = DataStreamReader()
         
         let requestOptions = HttpRequestOptions(request: crtRequest) { [self] (stream, _, httpHeaders) in
             logger.debug("headers were received")
-            response.statusCode = HttpStatusCode(rawValue: Int(stream.getResponseStatusCode()))
-                ?? HttpStatusCode.notFound
+            response.statusCode = HttpStatusCode(rawValue: Int(stream.statusCode))
+            ?? HttpStatusCode.notFound
             response.headers.addAll(httpHeaders: httpHeaders)
         } onIncomingHeadersBlockDone: { [self] (stream, _) in
             logger.debug("header block is done")
-            response.statusCode = HttpStatusCode(rawValue: Int(stream.getResponseStatusCode()))
-                ?? HttpStatusCode.notFound
+            response.statusCode = HttpStatusCode(rawValue: Int(stream.statusCode))
+            ?? HttpStatusCode.notFound
         } onIncomingBody: { [self] (_, data) in
             logger.debug("incoming data")
             
             let byteBuffer = ByteBuffer(data: data)
             streamReader.write(buffer: byteBuffer)
-        } onStreamComplete: { [self] (_, error) in
+        } onStreamComplete: { [self] (stream, error) in
             logger.debug("stream completed")
             streamReader.hasFinishedWriting = true
             if case let CRTError.crtError(unwrappedError) = error {
                 if unwrappedError.errorCode != 0 {
                     logger.error("Response encountered an error: \(error)")
                     streamReader.onError(error: ClientError.crtError(error))
-                    future.fail(error)
-                    return
+                    continuation.resume(throwing: error)
                 }
             }
-
+            
             response.body = .stream(.reader(streamReader))
-            future.fulfill(response)
+            self.logger.debug("response came back with success")
+            if let stream = stream {
+                response.statusCode = HttpStatusCode(rawValue: Int(stream.statusCode)) ?? HttpStatusCode.notFound
+            }
+            continuation.resume(returning: response)
```

Comment:
it's unclear what the success cases and failure cases are -- seems like there ought to be a better way to write this so that we can clearly see what cases we should bail, and what cases we should succeed.

Reply:
okay happy to explore options with you...

Reply:
I re-read this code, and i think the only thing that is confusing is that we were not setting the `response.statusCode` in the `onIncomingBody` block nor the `onStreamComplete` body.
1.  Why is it now needed to set `response.statusCode` within the `onStreamComplete` block?
2. if it is needed, is there an else clause that needs to be filled out?  For example:
```
if let stream = stream {
    response.statusCode = ...
} else {
   response.statusCode = //does this need to be set?
}
```

Reply:
investigated this, first of all looks like it was just a miss for why we weren't originally setting it additionally in onIncomingBody (looks like we set it in all other callbacks), secondly, there was no need for the stream to be optional in the crt code in onStreamComplete. I have updated both now. Thanks for the catch!